Add performance job

cherrycl opened the pull request:
Signed-off-by: Cherry Wang <cherry@iotechsys.com>

1. Influxdb host, 3 test clients and jmeter node are defined by parameter.
2. InfluxDB host, first test client and jmeter node are required. Whether Second and third test client are enabled that depends on testcases quantities.

cherrycl committed:
Add performance testing job

Signed-off-by: Cherry Wang <cherry@iotechsys.com>

jamesrgregg reviewed (commented):
Have you tested this in the sanbox yet?

eb-oss commented on jjb/performance-testing/performance-test.yaml:
```diff
@@ -0,0 +1,22 @@
+
+- project:
+    name: performance-testing
+    project-name: performance-testing
```
There is currently a "performance-test" repo in edgexfoundry-holding, but it does not have a Jenkinsfile.

cherrycl replied:
Sorry, I didn't point out the Jenkinsfile is on reviewing, please see branch [issue-7](https://github.com/edgexfoundry-holding/performance-test/tree/issue-7).

eb-oss commented on jjb/edgex-templates-performance.yaml:
```diff
@@ -0,0 +1,227 @@
+          jenkins-ssh-credential: '{jenkins-ssh-credential}'
+          submodule-timeout: 10
+
+    #triggers:
```
Is the plan to remove all of this commented-out code once testing is complete, before merging? (this applies to the several sections below as well; I'll just put the comment here)

cherrycl replied:
Yes, I will remove the commented code after testing is complete on sandbox. Thanks.

cherrycl commented:
> Have you tested this in the sanbox yet?
@jamesrgregg  I didn't test in the sandbox. I don't have authorization. Does anybody help this?

jamesrgregg commented:
@cherrycl I’ll request sandbox access for you.

jamesrgregg commented:
@cherrycl good job getting the job landed on the sandbox.  It looks like the job is failing because it needs docker-compose installed.  We want to see a successful test with data stored up to the InfluxDB before we accept this PR.

jamesrgregg commented:
@cherrycl are you ready to get this moved out of edgexfoundry-holding?

cherrycl commented:
@jamesrgregg Do you mean moving performance test repository from edgexfoundry-holding to edgexfoundry? I just changed the url of repository to edgexfoundry in jjb file. Once the performance test move to edgexfoundry github, it should be worked.

anonymouse64 committed:
Fix typos in edgex-template-{go,java,snap}.yaml

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

anonymouse64 committed:
Remove redundant "snap-" in arm verify job name

Previously this read in PR's as "edgex-go-snap-snap-master-verify-arm"
Now it will be "edgex-go-snap-master-verify-arm"

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

anonymouse64 committed:
Add new snap build script for snap jobs

This script is meant to build snaps and is combined from the current
files in edgex-go, but generalized to work with any repo with a
snapcraft.yaml in it.

We use this script as the build_script for the default jenkins job
boiler plate for snap job templates so that simply defining the project
and streams for a new repo with appropriately named jobs will enable
snap builds for new repos.

Note that we have to currently store both the Dockerfile and the
entrypoint.sh script inside the edgexfoundry-snapcraft.sh script using
heredocs because we need those things to be available to the job when it
runs and it's easier to store them inside this single script which gets
injected into the job definition rather than try to pull those files
from somewhere else when the job runs.

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

anonymouse64 committed:
jjb/ui/edgex-ui-go.yaml: use custom go on master job definition

This allows go modules to work properly

Fixes #356

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

Commit:
Add github webhook workaround for pipeline jobs

Signed-off-by: Lisa Rashidi-Ranjbar <lisa.a.rashidi-ranjbar@intel.com>

anonymouse64 committed:
Adjust edgexfoundry-snapcraft.sh comments

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

vishwassp-hcl-intel committed:
Create JJB views to filter Pipeline Jobs

Signed-off-by: Vishwas SP <vishwasx.s.premachandra@intel.com>

anonymouse64 committed:
sha512 check the snaps, download core18 in snap jobs

* Downloading the core18 snap lets snapcraft correctly check lib
  dependencies in the built snap against what is in the base (core18)
  snap and error if there are dynamic executables in the built snap that
  have library dependencies not satisfied by files in the built snap and
  not in the base snap.
* Hash the downloaded snap files using the SHA512 hash the store API
  provides in the JSON to prevent MITM attacks of the core or snapcraft
  snaps content (unlikely) and to error out early if network errors
  result in the file being corrupted during download (very likely)

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

ernestojeda committed:
remove docker job after moving to ci-build-images

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

Commit:
Change go-mod-messaging jobs to pipeline jobs

Signed-off-by: Lisa Rashidi-Ranjbar <lisa.a.rashidi-ranjbar@intel.com>

Commit:
Change go-mod-registry jobs to pipelines

Signed-off-by: Lisa Rashidi-Ranjbar <lisa.a.rashidi-ranjbar@intel.com>

ernestojeda committed:
add new base build image for kong-arm

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

ernestojeda committed:
removed hyphens causing jobs not to show

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

anonymouse64 committed:
Add device-mqtt snap jobs

This adds verify, stage, and release jobs for device-mqtt for the snap.
Currently only enable for master branch, we won't release the snap for
delhi.

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

anonymouse64 committed:
Add device-modbus snap jobs

This adds verify, stage, and release jobs for device-modbus for the
snap. Currently only enable for master branch, we won't release the snap
for delhi.

Signed-off-by: Ian Johnson <ian.johnson@canonical.com>

eb-oss committed:
Update centos7-builder image, remove basebuild

centos7-builder and centos7-basebuild were both very out of date.
Builder is meant to replace basebuild, and basebuild is not in use
in any job, so it has been entirely removed from the repo.

Signed-off-by: Eric Ball <eball@linuxfoundation.org>

jamesrgregg reviewed (approved):
LGTM

eb-oss reviewed (changes requested):
This needs to be properly rebased so that only the changes unique to this branch show up. Also, my feedback has not been addressed: references to `performance-testing` need to be changed to `performance-test`, and commented-out code needs to be removed.
